From 7859f9ba3ca24eb38fa8c3c03d5d311e6252a7c1 Mon Sep 17 00:00:00 2001
From: Nikolai Kondrashov <spbnick@gmail.com>
Date: Fri, 26 Nov 2010 23:34:30 +0300
Subject: [PATCH 2/4] Improve dumb tablet detection

Improve dumb tablet detection by checking if BTN_STYLUS or BTN_STYLUS2
(first or second pen barrel button) is present when BTN_TOOL_PEN is not
found.

This enables detection of tablets without explicit tool proximity reporting.
These include at least four UC-Logic tablet models: WP4030U, WP5540U,
WP8060U and PF1209, on which many Genius and Trust models are based.

Signed-off-by: Nikolai Kondrashov <spbnick@gmail.com>
---
 src/evdev.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/src/evdev.c b/src/evdev.c
index cd833dd..cccba97 100644
--- a/src/evdev.c
+++ b/src/evdev.c
@@ -1927,7 +1927,9 @@ EvdevProbe(InputInfoPtr pInfo)
         if ((TestBit(ABS_X, pEvdev->abs_bitmask) &&
              TestBit(ABS_Y, pEvdev->abs_bitmask))) {
             xf86Msg(X_INFO, "%s: Found x and y absolute axes\n", pInfo->name);
-            if (TestBit(BTN_TOOL_PEN, pEvdev->key_bitmask))
+            if (TestBit(BTN_TOOL_PEN, pEvdev->key_bitmask) ||
+                TestBit(BTN_STYLUS, pEvdev->key_bitmask) ||
+                TestBit(BTN_STYLUS2, pEvdev->key_bitmask))
             {
                 xf86Msg(X_INFO, "%s: Found absolute tablet.\n", pInfo->name);
                 pEvdev->flags |= EVDEV_TABLET;
-- 
1.7.2.3

